crates/sui/src/keytool.rs

```diff
@@ -351,6 +376,100 @@ impl KeyToolCommand {
                     }
                 }
             }
+            KeyToolCommand::SignKMS {
+                data,
+                keyid,
+                intent,
+            } => {
+                // Currently only supports secp256k1 keys
+                println!("Raw tx_bytes to execute: {}", data);
+                let intent = intent.unwrap_or_else(Intent::sui_transaction);
+                println!("Intent: {:?}", intent);
+                let msg: TransactionData =
+                    bcs::from_bytes(&Base64::decode(&data).map_err(|e| {
+                        anyhow!("Cannot deserialize data as TransactionData {:?}", e)
+                    })?)?;
+                let intent_msg = IntentMessage::new(intent, msg);
+                println!(
+                    "Raw intent message: {:?}",
+                    Base64::encode(bcs::to_bytes(&intent_msg)?)
+                );
+                let mut hasher = DefaultHash::default();
+                hasher.update(bcs::to_bytes(&intent_msg)?);
+                let digest = hasher.finalize().digest;
+                println!("Digest to sign: {:?}", Base64::encode(digest));
+
+                // Set up the KMS client, expect to use env vars or whatever the AWS
+                // SDK prefers.
+                let region: Region = Region::default();
+                let kms: KmsClient = KmsClient::new(region);
+
+                // Construct the signing request
+                let request: SignRequest = SignRequest {
+                    key_id: keyid.to_string(),
+                    message: digest.to_vec().into(),
+                    message_type: Some("RAW".to_string()),
+                    signing_algorithm: "ECDSA_SHA_256".to_string(),
+                    ..Default::default()
+                };
+                // Sign the message, normalize the signature and then compacts it
+                // serialize_compact is loaded as bytes for Secp256k1Sinature
+                // to convert it into a RecoverableSignature
+
+                let response = kms.sign(request).await?;
+                let sig_bytes_der = response
+                    .signature
+                    .map(|b| b.to_vec())
+                    .expect("Requires Asymmetric Key Generated in KMS");
+
+                let mut sig = Secp256k1Sig::from_der(&sig_bytes_der)?;
+                sig.normalize_s();
+                let sig_compact: &[u8; 64] = &sig.serialize_compact();
+                let fc_secp256k1sig = Secp256k1Signature::from_bytes(sig_compact)?;
+
+                let mut n_bytes: [u8; 65] = [0u8; 65];
+                n_bytes[..64].copy_from_slice(&sig_compact[..]);
+                let mut secppk: Option<Secp256k1PublicKey> = None;
```

Comment:
this is a very odd way to initialize in rust. consider use result and handle error instead of using option